lib/structs/events/collections.cpp

```diff
@@ -330,6 +331,10 @@ from_json(const nlohmann::json &obj, TimelineEvent &e)
                 break;
             }
             case MsgType::Unknown: {
+                e.data = events::RoomEvent<events::msg::Unknown>(obj);
+                break;
+            }
+            case MsgType::Invalid: {
                 e.data = events::RoomEvent<events::Unknown>(obj);
```

Comment:
Does this still work, since you've changed what an 'Unknown' event is from a JSON perspective?

Reply:
Yes, this works. `events::Unknown` still means anything that is potentially valid but that we don't know how to parse it. `events::msg::Unknown` means any `m.room.message` event (as represented by the `msg` sub-namespace) that is valid but that we don't understand the `msgtype` for.